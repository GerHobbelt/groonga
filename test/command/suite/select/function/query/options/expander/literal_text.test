table_create Memos TABLE_NO_KEY
column_create Memos content COLUMN_SCALAR ShortText

table_create Lexicon TABLE_PAT_KEY ShortText \
  --normalizer NormalizerAuto \
  --default_tokenizer TokenBigram
column_create Lexicon memos_content COLUMN_INDEX|WITH_POSITION Memos content

load --table Memos
[
["content"],
["Start Groonga!"],
["Start Mroonga!"],
["Start Rroonga!"],
["Start Ruby!"],
["Learning Ruby and Groonga..."]
]

select \
  --table Memos \
  --filter 'query("content", "rroonga content:@rroonga", \
                  {"expander": {"rroonga": "Ruby groonga"}})'
