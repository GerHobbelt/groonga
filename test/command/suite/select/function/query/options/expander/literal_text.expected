table_create Memos TABLE_NO_KEY
[[0,0.0,0.0],true]
column_create Memos content COLUMN_SCALAR ShortText
[[0,0.0,0.0],true]
table_create Lexicon TABLE_PAT_KEY ShortText   --normalizer NormalizerAuto   --default_tokenizer TokenBigram
[[0,0.0,0.0],true]
column_create Lexicon memos_content COLUMN_INDEX|WITH_POSITION Memos content
[[0,0.0,0.0],true]
load --table Memos
[
["content"],
["Start Groonga!"],
["Start Mroonga!"],
["Start Rroonga!"],
["Start Ruby!"],
["Learning Ruby and Groonga..."]
]
[[0,0.0,0.0],5]
select   --table Memos   --filter 'query("content", "rroonga content:@rroonga",                   {"expander": {"rroonga": "Ruby groonga"}})'
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        1
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "content",
          "ShortText"
        ]
      ],
      [
        5,
        "Learning Ruby and Groonga..."
      ]
    ]
  ]
]
